internal/k8s/rancher/rancher_desktop.go

```diff
@@ -0,0 +1,135 @@
+package rancher
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"runtime"
+	"strings"
+
+	"github.com/adrg/xdg"
+
+	"github.com/tilt-dev/tilt/pkg/logger"
+)
+
+type ContainerRuntime int
+
+const (
+	ContainerRuntimeUnknown ContainerRuntime = iota
+	ContainerRuntimeContainerd
+	ContainerRuntimeDocker
+)
+
+// openFileFunc is a hack for testing.
+type openFileFunc func(name string) (io.ReadCloser, error)
+
+var openFile = func(name string) (io.ReadCloser, error) {
+	return os.Open(name)
+}
```

Comment:
Do you need the wrapper?
```suggestion
var openFile = os.Open
```

Reply:
Ooo thanks, Go was being annoying about types here originally